### ISSUE TITLE
Request ran for longer than 1.8 seconds

### ISSUE THREAD

**Issue by mrcasals** [heroku/rack-timeout#66]
Hi! 

I'm using `rack-timeout 0.2.0` on Heroku with Rails.I get this error from time to time:

```
2015-03-12T08:30:11.110453+00:00 app[web.2]: E, [2015-03-12T08:30:11.110420 #12] ERROR -- : /app/vendor/bundle/ruby/2.2.0/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:670:in `worker_loop'
2015-03-12T08:30:11.110484+00:00 app[web.2]: E, [2015-03-12T08:30:11.110450 #12] ERROR -- : /app/vendor/bundle/ruby/2.2.0/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:525:in `spawn_missing_workers'
2015-03-12T08:30:11.110514+00:00 app[web.2]: E, [2015-03-12T08:30:11.110481 #12] ERROR -- : /app/vendor/bundle/ruby/2.2.0/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:140:in `start'
2015-03-12T08:30:11.110577+00:00 app[web.2]: E, [2015-03-12T08:30:11.110512 #12] ERROR -- : /app/vendor/bundle/ruby/2.2.0/gems/unicorn-4.8.3/bin/unicorn:126:in `<top (required)>'
2015-03-12T08:30:11.110611+00:00 app[web.2]: E, [2015-03-12T08:30:11.110576 #12] ERROR -- : /app/vendor/bundle/ruby/2.2.0/bin/unicorn:23:in `load'
2015-03-12T08:30:11.110640+00:00 app[web.2]: E, [2015-03-12T08:30:11.110608 #12] ERROR -- : /app/vendor/bundle/ruby/2.2.0/bin/unicorn:23:in `<main>'
2015-03-12T08:30:11.111314+00:00 app[web.2]: source=rack-timeout id=84784c36-ceb1-41cb-a6c4-d0a8fbfb7761 wait=30696ms timeout=30000ms state=expired
2015-03-12T08:30:11.160040+00:00 app[web.2]:
2015-03-12T08:30:11.160044+00:00 app[web.2]: ActionView::Template::Error (Request ran for longer than 1.8885111324057604 seconds.):
```

So the request ran for 30696ms when the timeout is 30000ms, but the error says it ran for longer than 1.8 seconds, which is wrong.

Also, a weird part is that I have this in an initializer:

```ruby
Rack::Timeout.timeout = 22 if Rails.env.production?
```

Am I doing something wrong, or did I catch a bug?

Thanks!

**Comment by kch** [comment]
That error message is for a `RequestTimeoutError` (but rails is lame and wraps errors raised during view code in an `ActionView` error)… the line above says the `state` is `expired`, so that would have been a `RequestExpiryError`. And these are dropped before they ever reach rails so ActionView would have never been in play.

So basically the two lines are talking about different requests. You should have a log line somewhere else with `state=timed_out` and `service=18???ms` that correlates with the raised error you see.

As for timing out at 1.8 instead of 22, that's related to the wait time. You can read more about it in the readme.

**Comment by kch** [comment]
Also I don't think you're running 0.2.0, otherwise the "ran for longer" message would have been different, secs in ms, wait listed.

**Comment by mrcasals** [comment]
Yes, I double checked the version and I'm using 0.2.0:

```
$ cat Gemfile.lock | grep rack-timeout
    rack-timeout (0.2.0)
  rack-timeout
```

Also, I've searched the logs and found some lines with `state=timed_out`:

```
Mar 12 01:18:33 empresaula app/web.1:  source=rack-timeout id=bec2ca50-682d-4c8e-8314-ae59ec7aa534 wait=28790ms timeout=1210ms service=1215ms state=timed_out 
Mar 12 01:18:33 empresaula app/web.1:  source=rack-timeout id=3870d089-3e71-4c47-ae6c-95c3599fb4b1 wait=29391ms timeout=609ms service=611ms state=timed_out 
Mar 12 01:18:34 empresaula app/web.1:  source=rack-timeout id=583ba945-f715-4797-8f4c-8d3023dfeba6 wait=29857ms timeout=143ms service=151ms state=timed_out 
Mar 12 01:18:47 empresaula app/web.1:  source=rack-timeout id=0ddcf58b-ae0b-49a6-8cb3-4ab793ce76bf wait=29776ms timeout=224ms service=224ms state=timed_out 
```

Taking, for example, the last line, I guess this is what happens: request runs for 29776ms before it gets any data (wait time). Then the server starts processing for 224ms, but as 29776 + 224 = 30000ms, request is timed-out. So the fault of these "Request ran for longer than 1.8 seconds" is the wait time, which is too high. Is that correct?

**Comment by kch** [comment]
Correct. Also this wait time situation seems terrible. You should contact heroku support and see if we can help you with that.

As for you being on 2.0, it's really weird that you're getting that message in that particular form (it doesn't matter though, it's just a string). I wonder if I botched things when I released 2.0. Will check. I put a 2.1 out anyhow.
